pcsx2-qt/ShortcutCreationDialog.cpp

```diff
@@ -40,404 +51,207 @@ ShortcutCreationDialog::ShortcutCreationDialog(QWidget* parent, const QString& t
+	                           .toStdString();
 
-	std::string cmdline = StringUtil::JoinString(passed_cli_args.begin(), passed_cli_args.end(), " ");
+	const std::string shortcut_exec = fmt::format("{} {} -- {}", executable_path, StringUtil::JoinString(standard_arguments.begin(), standard_arguments.end(), " "), m_game_path);
 
-	if (!is_flatpak)
+	if (!FileSystem::CreateLinuxDesktopFile(shortcut_destination, shortcut_exec, m_game_title, shortcut_comment))
 	{
-		// Copy PCSX2 icon
-		std::string icon_dest;
-		if (xdg_data_home)
-			icon_dest = fmt::format("{}/icons/hicolor/512x512/apps/", xdg_data_home);
-		else
-			icon_dest = fmt::format("{}/.local/share/icons/hicolor/512x512/apps/", home);
-
-		std::string icon_name = "PCSX2.png";
-		std::string icon_path = fmt::format("{}/{}", icon_dest, icon_name).c_str();
-		if (FileSystem::EnsureDirectoryExists(icon_dest.c_str(), true))
-			FileSystem::CopyFilePath(Path::Combine(EmuFolders::Resources, "icons/AppIconLarge.png").c_str(), icon_path.c_str(), false);
+		QMessageBox::critical(this, tr("Failed to create shortcut"), tr("Could not create .desktop file."),
+			QMessageBox::StandardButton::Ok, QMessageBox::StandardButton::Ok);
 	}
-
-	// Further string sanitization
-	if (!is_flatpak)
-		ShortcutCreationDialog::EscapeShortcutCommandLine(&executable_path);
-	ShortcutCreationDialog::EscapeShortcutCommandLine(&clean_path);
-
-	// Assembling the .desktop file
-	std::string final_args;
-	final_args = fmt::format("{} {} {} -- {}", executable_path, cmdline, custom_args, clean_path);
-	std::string file_content =
-		"[Desktop Entry]\n"
-		"Encoding=UTF-8\n"
-		"Version=1.0\n"
-		"Type=Application\n"
-		"Terminal=false\n"
-		"StartupWMClass=PCSX2\n"
-		"Exec=" + final_args + "\n"
-		"Name=" + clean_name + "\n"
-		"Icon=net.pcsx2.PCSX2\n"
-		"Categories=Game;Emulator;\n";
-	std::string_view sv(file_content);
-
-	// Prompt user for shortcut saving destination
-	QString final_path(QStringLiteral("%1").arg(QString::fromStdString(link_path)));
-	const QString filter(tr("Desktop Shortcut Files (*.desktop)"));
-
-	final_path = QDir::toNativeSeparators(QFileDialog::getSaveFileName(this, tr("Select Shortcut Save Destination"), final_path, filter));
-
-	if (final_path.isEmpty())
-		return;
-
-	// Write to .desktop file
-	if (!FileSystem::WriteStringToFile(final_path.toStdString().c_str(), sv))
-	{
-		QMessageBox::critical(this, tr("Failed to create shortcut"), tr("Failed to create .desktop file"), QMessageBox::StandardButton::Ok, QMessageBox::StandardButton::Ok);
-		return;
-	}
-
-	if (chmod(final_path.toStdString().c_str(), S_IRWXU) != 0) // enables user to execute file
-		Console.ErrorFmt("Failed to change file permissions for .desktop file: {} ({})", strerror(errno), errno);
 #endif
 }
 
-bool ShortcutCreationDialog::EscapeShortcutCommandLine(std::string* arg)
+#if defined(_WIN32)
+void ShortcutCreationDialog::EscapeCommandLineArgumentWindows(std::string& cli_argument)
```

Comment:
Please keep the pointer, it forces an `&` on the callsite to indicate that the function modifies its input

Also, making variable names longer does make things harder to read, so only do it if it actually helps understanding.  (IMO it doesn't here, given that you're already in a function called `EscapeCommandLineArgumentWindows`.)